plugins/source/aws/resources/services/codebuild/projects.go

```diff
@@ -43,24 +42,20 @@ func fetchCodebuildProjects(ctx context.Context, meta schema.ClientMeta, parent
 	c := meta.(*client.Client)
 	svc := c.Services().Codebuild
 	config := codebuild.ListProjectsInput{}
-	for {
-		response, err := svc.ListProjects(ctx, &config)
+	paginator := codebuild.NewListProjectsPaginator(svc, &config)
+	for paginator.HasMorePages() {
+		page, err := paginator.NextPage(ctx)
 		if err != nil {
 			return err
 		}
-		if len(response.Projects) == 0 {
-			break
+		if len(page.Projects) == 0 {
+			continue
```

Comment:
This page being empty doesn't mean we don't have more pages I'm guessing... so we leave the actual check to `paginator.HasMorePages()`

Reply:
Exactly

Reply:
Also we can't use the PreResourceResolver pattern here because https://github.com/cloudquery/plugin-sdk/issues/338